services/user/Fractals/query-service/src/lib.rs

```diff
@@ -0,0 +1,292 @@
+            EventQuery::new("history.evaluations.new_group")
+                .condition(format!(
+                    "owner = '{}' AND evaluation_id = {}",
+                    evaluation_owner, evaluation_id
+                ))
+                .first(first)
+                .last(last)
+                .before(before)
+                .after(after)
+                .query()
+        }
+
+        async fn evaluation_finishes(
+            &self,
+            fractal: AccountNumber,
+            first: Option<i32>,
+            last: Option<i32>,
+            before: Option<String>,
+            after: Option<String>,
+        ) -> async_graphql::Result<Connection<u64, EvaluationFinish>> {
+            EventQuery::new("history.fractals.evaluation_finished")
+                .condition(format!("fractal_account = '{}'", fractal))
+                .first(first)
+                .last(last)
+                .before(before)
+                .after(after)
+                .query()
+        }
+
+        async fn scheduled_evaluations(
+            &self,
+            fractal: AccountNumber,
+            first: Option<i32>,
+            last: Option<i32>,
+            before: Option<String>,
+            after: Option<String>,
+        ) -> async_graphql::Result<Connection<u64, ScheduledEvaluation>> {
+            EventQuery::new("history.fractals.scheduled_evaluation")
+                .condition(format!("fractal_account = '{}'", fractal))
+                .first(first)
+                .last(last)
+                .before(before)
+                .after(after)
+                .query()
+        }
+
+        async fn group_finishes(
+            &self,
+            evaluation_id: u32,
+            first: Option<i32>,
+            last: Option<i32>,
+            before: Option<String>,
+            after: Option<String>,
+        ) -> async_graphql::Result<Connection<u64, GroupFinish>> {
+            EventQuery::new("history.evaluations.group_fin")
+                .condition(format!(
+                    "owner = 'fractals' AND evaluation_id = {}",
+                    evaluation_id
+                ))
+                .first(first)
+                .last(last)
+                .before(before)
+                .after(after)
+                .query()
+        }
+
+        async fn fractal(&self, fractal: AccountNumber) -> Option<Fractal> {
+            FractalTable::with_service(fractals::SERVICE)
+                .get_index_pk()
+                .get(&fractal)
+        }
+
+        async fn evaluations(&self, fractal: AccountNumber) -> Vec<EvaluationInstance> {
+            EvaluationInstanceTable::with_service(fractals::SERVICE)
+                .get_index_pk()
+                .range((fractal, 0)..=(fractal, u32::MAX))
+                .collect()
+        }
+
+        async fn scores(
+            &self,
+            fractal: AccountNumber,
+            member: Option<AccountNumber>,
+        ) -> Vec<Score> {
```

Comment:
`get_scores` should be a paginated query that returns a subindex (See r-evaluations::get_groups for example)
And there should be a separate query for `get_score` that takes a member and returns just their one score record

Reply:
All users have two scores, one for rep and another for favour. 
Do I just use a range? 
New version updated now is using TableQuery::subindex